src/app/CASESessionManager.cpp

```diff
@@ -30,7 +30,8 @@ CHIP_ERROR CASESessionManager::Init(chip::System::Layer * systemLayer, const CAS
 }
 
 void CASESessionManager::FindOrEstablishSession(const ScopedNodeId & peerId, Callback::Callback<OnDeviceConnected> * onConnection,
-                                                Callback::Callback<OnDeviceConnectionFailure> * onFailure
+                                                Callback::Callback<OnDeviceConnectionFailure> * onFailure,
+                                                Callback::Callback<OnExtendedDeviceConnectionFailure> * onExtendedConnectionFailure
```

Comment:
I believe @tcarmelveilleux was suggesting we have a new FindOrEstablishSession with a struct argument instead of continuing to add arguments, and make the old API call that.

In any case, this is breaking the existing API, which we should not be doing.  We should be adding another FindOrEstablishSession overload instead.